src/core/mesh/qgsmeshdataset.h

```diff
@@ -547,6 +556,19 @@ class CORE_EXPORT QgsMeshDatasetGroupTreeItem
 {
   public:
 
+    /**
+     * Where the dataset group is stored
+     *
+     * \since QGIS 3.16
+     */
+    enum StorageType
+    {
+      None, //! Generic item that do not store dataset
+      File, //! Dataset group store on file
+      Memory, //! Temporary dataset group in memory
+      OnTheFly //! dataset group depending on oahter dataset and that is calculated when needed
```

Comment:
dataset group depending on oahter dataset and that is calculated when needed -> Dataset Group which is based on expression evaluated in real-time